utils/k8s.go

```diff
@@ -0,0 +1,178 @@
+	}
+
+	return clientSet, dynClient, nil
+}
+
+func AuthenticateOutOfCluster() (*kubernetes.Clientset, dynamic.Interface, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get client config: %v", err)
+	}
+
+	// create the clientset
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate client set: %v", err)
+	}
+
+	// creates dynClient for kube watching
+	dynClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate dynamic client : %v", err)
+	}
+
+	return clientSet, dynClient, nil
+}
+
+func GetKubeStuckNodesInfos(clientSet *kubernetes.Clientset) []FullNode {
+	nodes, err := clientSet.CoreV1().Nodes().List(context.TODO(), v1.ListOptions{Limit: int64(1000)})
+
+	if err != nil {
+		log.Errorf("Can't list kubernetes nodes: %s", err.Error())
+	}
+
+	stuckNodes := getStuckNodes(nodes)
+
+	if len(stuckNodes) == len(nodes.Items) {
+		log.Debug("Cluster isn't ready yet.")
+		return []FullNode{}
+	}
+
+	return stuckNodes
+}
+
+/**
+* Using three conditions to check if node is stock : node Ready status is at "unknown", it's not Unschedulable and it was
+* created more than five minutes ago.
+ */
+func checkNode(node v12.Node) FullNode{
+	var fullNode FullNode
+
+	for _, condition := range node.Status.Conditions {
+		if strings.EqualFold(fmt.Sprint(condition.Type), "Ready") &&
+			strings.EqualFold(fmt.Sprint(condition.Status), "Unknown") &&
+			!node.Spec.Unschedulable &&
+			node.CreationTimestamp.Add(time.Minute *5).Before(time.Now()) {
+			return getNodesInfos(node)
+		}
+	}
+
+	return fullNode
+}
+
+func getStuckNodes(nodelist *v12.NodeList) []FullNode {
+	var stuckNodesId []FullNode
+
+	for _, node := range nodelist.Items {
+		checkedNode := checkNode(node)
+		stuckNodesId = append(stuckNodesId, checkedNode)
+	}
+
+	return stuckNodesId
+}
+
+func WatchNodes(clientSet *kubernetes.Clientset, dynClinent dynamic.Interface, DOclient *godo.Client) {
+	result, err := clientSet.CoreV1().Nodes().Watch(context.TODO(), v1.ListOptions{Limit: int64(1000)})
+	if err != nil {
+		log.Printf("Can't watch nodes: %s", err)
+	}
+
+
+	for event := range result.ResultChan() {
+		if event.Type == watch.Added || event.Type == watch.Modified {
+			node := getNodesInfosFromRuntineObject(dynClinent, "default", event.Object)
+			checkedNode := checkNode(node)
+			emptyNode := FullNode{}
+			if checkedNode != emptyNode {
+				//RecyleNode(DOclient, checkedNode)
+			}
+		}
+	}
+}
+
+func getNodesInfosFromRuntineObject(dynClient dynamic.Interface, nameSpace string, obj runtime.Object) v12.Node{
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		log.Errorf("Can't unstructure object: %s", err)
```

Comment:
Please give more context (ex: kind of object)

Reply:
done